src/lightning/pytorch/utilities/argparse.py

```diff
@@ -15,101 +15,36 @@
 
 import inspect
 import os
-from argparse import _ArgumentGroup, ArgumentParser, Namespace
+from argparse import Namespace
 from ast import literal_eval
 from contextlib import suppress
 from functools import wraps
-from typing import Any, Callable, cast, Dict, List, Tuple, Type, TypeVar, Union
+from typing import Any, Callable, cast, Type, TYPE_CHECKING, TypeVar, Union
 
-import lightning.pytorch as pl
-from lightning.pytorch.utilities.parsing import str_to_bool, str_to_bool_or_int, str_to_bool_or_str
-from lightning.pytorch.utilities.types import _ADD_ARGPARSE_RETURN
+if TYPE_CHECKING:
+    import lightning.pytorch as pl
```

Comment:
This is unused

```suggestion
```

Reply:
It was used for the definition of _ARGPARSE_CLS